react/features/video-layout/subscriber.js

```diff
@@ -12,6 +12,8 @@ import { setRemoteParticipantsWithScreenShare } from './actions';
 declare var APP: Object;
 declare var interfaceConfig: Object;
 
+let latestScreenshareParticipantId;
```

Comment:
Why can't we check it the pinned participant is still in the array instead?

Reply:
Yeah we could also do that.

Reply:
The `remoteScreenShares` array is empty because all the SS endpoints have left. We need to check if the pinned participant in redux is not the last SS endpoint that left.

Reply:
Can't we just check if the currently pinned participant is no longer in the conference, regardless of them sharing their screen previously?

Or am I missing something as to why this requires special handling?

Reply:
Let me add the context here: An endpoint starts sharing and get auto-pinned. Some endpoints(ep2) might override the auto-pin and pin another participant in the call. When the screen-sharing participant stops sharing or leaves the call, client needs to determine if the view should switch to tile view or should continue to show pinned participants in stage view for ep2. So here when all the remote shares have left, client needs to check if the pinned participant in redux is the SS ep or some other ep and act accordingly.

Reply:
This functionality got broken because of a previous commit and I am fixing the regression. (This is not the YT share issue)

Reply:
I see. The listener that calls this function has access to both the old and new arrays of screen-shares, so maybe we can leverage that?

Reply:
Sure, I pushed a commit that uses oldScreenSharesOrder.